packages/x-charts/src/internals/identifierSerializer.ts

```diff
@@ -0,0 +1,17 @@
+import type { SeriesId } from '../models';
+
+export const typeSerializer = (type: string) => `Type(${type})`;
+export const seriesIdSerializer = (id: SeriesId) => `Series(${id})`;
+export const dataIndexSerializer = (dataIndex?: number) => `Index(${dataIndex})`;
+
+export const identifierSerializerSeriesIdDataIndex = <
+  T extends {
+    type: string;
+    seriesId: SeriesId;
+    dataIndex?: number;
+  },
+>(
+  identifier: T,
+) => {
+  return `${typeSerializer(identifier.type)}${seriesIdSerializer(identifier.seriesId)}${dataIndexSerializer(identifier.dataIndex)}`;
+};
```

Comment:
I don't get why using a generic here

```suggestion
export const identifierSerializerSeriesIdDataIndex = (
  identifier: {
    type: string;
    seriesId: SeriesId;
    dataIndex?: number;
  },
) => {
  return `${typeSerializer(identifier.type)}${seriesIdSerializer(identifier.seriesId)}${dataIndexSerializer(identifier.dataIndex)}`;
};
```